package/ios/Core/CameraSession+Orientation.swift

```diff
@@ -63,10 +66,14 @@ extension CameraSession: OrientationManagerDelegate {
 
     VisionLogger.log(level: .info, message: "Updating Preview rotation: \(previewOrientation)...")
 
+    guard let device = videoDeviceInput?.device else {
+      return
+    }
+
     // update the orientation for each preview layer that is connected to this capture session
     let previewConnections = captureSession.connections.filter { $0.videoPreviewLayer != nil }
     for connection in previewConnections {
-      connection.orientation = previewOrientation
+      connection.setOrientation(newOrientation: previewOrientation, device: device)
```

Comment:
I don't think this is the right way to implement this.

I want `orientation` to be a simple property, not a setter/getter function that takes a `device`.

If we do any operations on `device` inside `setOrientation`, why do we not do that beforehand (e.g. in `OrientationManager` when we create `previewOrientation`)?

This is hard to read, so I think this should be the `OrientationManager`'s responsibility.

Can you refactor it that way?

Thank you!